### ISSUE TITLE
`mailto:` link cannot be detected by some MUAs

### ISSUE THREAD

**Issue by abkayla** [sympa-community/sympa#1124]
**Version**

6.2.40 on Ubuntu 20.04

**Installation method**

Ubuntu 20.04: From source

**Expected behavior**

mailto: links generated by mail_tt2/moderate.tt2 : should look something like:

<mailto:sympa@XXX.domain.fr?subject=DISTRIBUTE%20test-dpc99%20a29a51710113bb50991b18fb3e03ebb1>

**Actual behavior**
This doesn't work correctly when people click on the link.
and a message received to the people after  using the link with  

>Object: Result of your commands

> The command failed

> DISTRIBUTE XXX a29a51710113bb50991b18fb3e03ebb1>
>  List '' is unknown.

**Comment by ikedas** [comment]
Hi @khaoula-abaidi ,

Could you please show (attach) the e-mail itself that was generated?

**Comment by khaoula-abaidi** [comment]
Mail received for moderation:

![mail_received](https://user-images.githubusercontent.com/37698060/109121957-35dda180-7748-11eb-9114-a6e1f2cdf889.png)

Mail received after click to the link and send the mail:

![when_clicked_send_mail](https://user-images.githubusercontent.com/37698060/109122025-4db52580-7748-11eb-9b86-f5e67cc59466.png)


Note : 
If i delete the ">"  from the subject of the mail that must be sending after clicking to the link , the action is done , else i receive the 2nd message error

**Comment by ikedas** [comment]
@khaoula-abaidi , could you please tell the name and version of the MUA (mail client software) you are using?

**Comment by khaoula-abaidi** [comment]
> @khaoula-abaidi , could you please tell the name and version of the MUA (mail client software) you are using?

Using Zimbra i have this problem , but using Gmail it's more problematic because the link is shown just for the @mail not all the link
![gmail](https://user-images.githubusercontent.com/37698060/109127719-05e5cc80-774f-11eb-8889-ce2e8d6c8044.png)

**Comment by ikedas** [comment]
Though this is not a bug of Sympa itself, it'd be better to be dealt with by Sympa.

My idea is that adding web function do_confirm() on WWSympa  and ncluding its link in the messages.

**Comment by khaoula-abaidi** [comment]
> Though this is not a bug of Sympa itself, it'd be better to be dealt with by Sympa.
> 
> My idea is that adding web function do_confirm() on WWSympa and ncluding its link in the messages

Would you please specify how to include the link in the messages ?

**Comment by ikedas** [comment]
> Would you please specify how to include the link in the messages ?

Please wait for a while.  It needs modification to the code.

**Comment by khaoula-abaidi** [comment]
> > Would you please specify how to include the link in the messages ?
> 
> Please wait for a while. It needs modification to the code.

Will you please specify if the corrections will be into sympa 6.2.40 version and approximately the date of update of the corrections

**Comment by ikedas** [comment]
I tihnk 6.2.40 may not be fixed.  I'm planning to include the fix in the future verskion of Sympa.  Ubuntu may adopt that fixed version or later version as the upstream version in the future release.

**Comment by khaoula-abaidi** [comment]
> I tihnk 6.2.40 may not be fixed. I'm planning to include the fix in the future verskion of Sympa. Ubuntu may adopt that fixed version or later version as the upstream version in the future release.

So all users of Sympa before the future version must upgrade sympa to get new corrections.

**Comment by ikedas** [comment]
> > I tihnk 6.2.40 may not be fixed. I'm planning to include the fix in the future verskion of Sympa. Ubuntu may adopt that fixed version or later version as the upstream version in the future release.
> 
> So all users of Sympa before the future version must upgrade sympa to get new corrections ?

No, not all users.  Sympa is the free software (open source software).  You can modify it anyhow you want to be, by modifying the source code _by yourself_.  For example, it is possible that you take the fixes I'll make on future version and apply it to the version you are running.

Ubuntu project will not do such work for you unless there are good reasons to do so. (You can ask them to do it for you, but I think they will probably refuse.)  Instead, they will release an updated OS adopting a new version of Sympa as upstream.

**Comment by khaoula-abaidi** [comment]
I'm sorry for all my questions , but i want to tell you that the same problem is generated also for send_auth.tt2

**Comment by khaoula-abaidi** [comment]
Is it possible to tell me the delay for the corrections into the new version ? Thank you for your response

**Comment by ikedas** [comment]
Report above tells the problem has not been resolved with Outlook for Mac.  Reopen.

**Comment by ikedas** [comment]
The fix will be included in the next release (perhaps 6.2.71b). @abkayla , thank you for reporting bug!

**Comment by laerm** [comment]
Unless I'm mistaken, you've made the choice to hide the mailto option in moderation emails if conf.wwsympa_url is present? This is problematic for some of our users who are unable to log into the web interface (we use an SSO and have external moderators for certain lists). Would it be possible to add this as optional in sympa.conf file so we can turn on the mailto? We'd rather have a mailto that doesn't work in all MUAs than no mailto link whatsever.

**Comment by fusenuk** [comment]
We've also just been caught out by this change of hiding the mailto link. I also don't see it mentioned in the change log.

Please can this option be returned or/also please can this related issue be addressed - https://github.com/sympa-community/sympa/issues/965

Using the mailto was the workaround to the above 'bug'

**Comment by ikedas** [comment]
@laerm ,

One option is customizing template.  The changes we made are shown [here](https://github.com/sympa-community/sympa/pull/1507/files). You may customize the template to revert these changes (Note that the files under DEFAULTDIR should not be modified directly.  See [the manual](https://www.sympa.community/manual/customize/basics-templates.html) on how to customize the templates).

Another option is disabling web interface.  If the `wwsympa_url` parameter is not defined in `sympa.conf` (nor in `robot.conf`, if there is one), the notification will contain a `mailto:` URL instead of a link to the web interface.

**Comment by agouaux** [comment]
> Unless I'm mistaken, you've made the choice to hide the mailto option in moderation emails if conf.wwsympa_url is present? This is problematic for some of our users who are unable to log into the web interface (we use an SSO and have external moderators for certain lists). Would it be possible to add this as optional in sympa.conf file so we can turn on the mailto? We'd rather have a mailto that doesn't work in all MUAs than no mailto link whatsever.

yeah, we were bit by this upgrading to 6.2.72. The old behavior never gave us problems. Once I copy default/mail_tt2/moderate.tt2 to etc/mail_tt2, what changes do I need to make to regain the old behavior?

**Comment by ikedas** [comment]
The situation is the same for HTTP URLs. See #1971 .
